### PR TITLE
fix header inversion bug

### LINKED ISSUE #42962: Notifications menu and contact-search menu broken with certain predefined backgrounds
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps

1. Go to go to index.php/settings/user/theming
2. Select one of the following default offered backgrounds: All that make the font color go black
  - Pink bubbles
  - Yellow bricks
  - Lion rock hill
  - "the ducks"
  - soft floral
  - morning fog
4. Open the notification menu => Inverted color ❌ 
5. Open the contact search => menu opens in place of the button instead of a popover menu 💥 
Bildschirmfoto vom 2024-01-19 14-43-44

How it looks with any other background:

Bildschirmfoto vom 2024-01-19 14-43-35

### PR DESCRIPTION
**Pull request by szaimen**
Should fix https://github.com/nextcloud/server/issues/42962

I am not really satisfied with the solution but it is what it is...

### CODE REVIEW

File: core/css/header.scss [hunk 1]
```diff
@@ -172,7 +172,20 @@
 	/* Right header standard */
 	.header-right {
 		> .header-menu:not(.user-menu) {
-			filter: var(--background-image-invert-if-bright);
+			// For general
+			> .header-menu__trigger {
```
Comment (susnux) [inline core/css/header.scss:176]:
```suggestion
			> button {
```

this should work for all I think

Comment (szaimen) [reply core/css/header.scss:176]: I fear the assistant button is not a button but a link...

Comment (szaimen) [reply core/css/header.scss:176]: Also I would rather not risk using the button tag here...

**Review by susnux — APPROVED** [review]
works!

File: core/css/header.scss (hunk 1 above)
Comment (susnux) [reply core/css/header.scss:176]: Then we should probably adjust the two other that do not use `.header-menu__trigger`.
But for the moment your solution works fine!

Comment (ShGKme) [reply core/css/header.scss:176]: > Also I would rather not risk using the button tag here...

```css
button,
a {}
```

?

Comment (szaimen) [reply core/css/header.scss:176]: Even more risky...

Comment (susnux) [reply core/css/header.scss:176]: or even better only apply to `img` tag. Everything else should handle it correctly and we prevent future issues

**Review by ShGKme — APPROVED** [review]

**Comment by blizzz** [comment]
rebased

**Review by AndyScherzinger — APPROVED** [review]

**Comment by szaimen** [comment]
CI failure unrleated afaics

**Pull request merged**